File: 백준/Silver/1010. 다리 놓기/다리 놓기.js
const input = (
  process.platform === "linux"
    ? require("fs").readFileSync("/dev/stdin").toString()
    : require("fs").readFileSync("예제.txt").toString()
)
  .trim()
  .split("\n");

input.shift();

input.forEach((element) => {
  let [a, b] = element.split(" ");
  let re = solution(b, a);
  console.log(re);
});

function solution(n, r) {
  return Math.round(fac(n) / (fac(r) * fac(n - r)));
}

function fac(x) {
  if (x <= 1) return 1;
  else return x * fac(x - 1);
}

File: 백준/Silver/1010. 다리 놓기/다리 놓기.py
import sys

def cal(n, m):
    if n > m - n:
        n = m - n
    result = 1
    for i in range(n):
        result = result * (m - i) // (i + 1)
    return result

def main():
    input = sys.stdin.readline
    
    k = int(input())
    for _ in range(k):
        n, m = map(int, input().strip().split())
        print(int(cal(n, m)))
if __name__ == "__main__":
    main()